docs/devel/infrastructure-interaction.md

```diff
@@ -0,0 +1,85 @@
+
+Separating the volume to a first class resource instead of part of the pod definition gets you all sorts of cool stuff.  Couple of initial thoughts are that pod definitions become much more portable, since the infrastructure specific portions are abstracted out.  Access controls become much more reasonable, as you can easily do checks on creating the volume resource and between volume resources and pods.  More thoughts below...
+
+Is this a sane design and starting point when talking about how to solve storage and external ips? If its a sane design is there interest in trying to normalize the minion stuff? Obviously actually solving any of those problems have 1,000,000 other details, but is this a sane design to move forward with?
+
+**Design Restated**
+
+1. infrastructure makes functionality available to kube
+2. kube admin/user explicitly tells the system how to use those resources.
+3. 'something' pulls that explicit information and tries to implement the request.
+
+(man, that sounds exactly like the original kube design!)
+
+**Mapping minions, GCE load balancer, and GCE PDs today with this perspective**
+
+***Minions:*** minion discovery ideas are schizophrenic at best. We've got –machines and –minion_regexp. Which are mutually exclusive and obviously not coherent together. Trying to rationalize minions with this design proposal works great for the ‘real cloud provider model’ but it is admittedly not a nice admin experience without it.  I think we should get rid of --machines entirely.  To solve the admin experience we should plumb the ‘add a minion to etcd’ API through kubectl. I’m not quite sure how to make adding minions with kubectl work sanely with the regex and the list of minions from the cloud provider.  Maybe just disable it for certain cloud providers.  Maybe have “kubectl create/delete minion” merge with the regex.  Details.  Magic.
+
+***GCE load balancer:*** In this case it’s handled by ‘createExternalLoadBalancer=true in the service file.  This causes the apiserver to make an (async) push type call into gce.service.ForwardingRules.Insert().  I seems to me that external services like this should be modeled as a new resource type separate from the service and that instead of the apiserver pushing the update to the cloudprovider/gce another task should run which pulls from the apiserver and does the push.  We’re trying to get away from the push model, right?  But I think the new kube resource argument is best fought in something like the external IPs PR.  This is a high level thought.
```

Comment:
I think we need both push and pull, though I understand that calls out to a cloud provider are problematic when there isn't one.